.husky/pre-commit

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npx --no-install lint-staged && npm test
```

Comment:
+1 on lint per-commit, but I personally prefer running test per push.

I use commit to commit a "work in progress" history; I use push to indicate "this is ready". If a PR is "work in progress", it may fail certain tests in some commits. These intermediate commits may not require fixing the tests yet because the work is not finished.

Lint per commit has its value because we can omit the "fix lint" commits. Also, fixing lints is a much easier effort than fixing tests. 

I think having a per-commit rule for `npm run test` will slow down development time, defeating the purpose of "commit often". 

I would like to hear what others think as well.

Reply:
Thanks for the input - I also try to commit often, but I do those intermediate commits with the `-n`/`--no-verify` flag.

What would your thoughts be on pushing without running the tests?
I was going through `woocommerce/woocommerce-payments`, `woocommerce/woocommerce`, `woocommerce/woocommerce-admin` and `Automattic/jetpack`, `WordPress/gutenberg`, they all don't run the tests on pre-commit nor on pre-push.
I would be kinda sad maybe to not have the tests run, but since they are already running on CI, that would make things faster 🤷

Reply:
I personally think that WCS&T should continue to run test per-push. I think we should make sure tests pass before we push, so automating this makes sense.

Reply:
Changed in 92139cc8cb2eaf40e03eed83166a564cb01bd22b 👍